src/main/java/bio/terra/workspace/db/ResourceDao.java

```diff
@@ -0,0 +1,429 @@
+import bio.terra.workspace.service.resource.reference.ReferenceResource;
+import bio.terra.workspace.service.workspace.model.CloudPlatform;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+@Component
+public class ResourceDao {
+  private static final Logger logger = LoggerFactory.getLogger(ResourceDao.class);
+
+  /** SQL query for reading all columns from the resource table */
+  private static final String RESOURCE_SELECT_SQL =
+      "SELECT workspace_id, cloud_platform, resource_id, name, description, "
+          + "stewardship_type, resource_type, cloning_instructions, attributes,"
+          + " access, associated_app, assigned_user"
+          + " FROM resource WHERE workspace_id = :workspace_id ";
+
+  private static final RowMapper<DbResource> DB_RESOURCE_ROW_MAPPER =
+      (rs, rowNum) -> {
+        return new DbResource()
+            .workspaceId(UUID.fromString(rs.getString("workspace_id")))
+            .cloudPlatform(CloudPlatform.fromSql(rs.getString("cloud_platform")))
+            .resourceId(UUID.fromString(rs.getString("resource_id")))
+            .name(rs.getString("name"))
+            .description(rs.getString("description"))
+            .stewardshipType(fromSql(rs.getString("stewardship_type")))
+            .resourceType(WsmResourceType.fromSql(rs.getString("resource_type")))
+            .cloningInstructions(CloningInstructions.fromSql(rs.getString("cloning_instructions")))
+            .attributes(rs.getString("attributes"))
+            .accessType(
+                Optional.ofNullable(rs.getString("access"))
+                    .map(ControlledAccessType::fromSql)
+                    .orElse(null))
+            .associatedApp(
+                Optional.ofNullable(rs.getString("associated_app"))
+                    .map(UUID::fromString)
+                    .orElse(null))
+            .assignedUser(rs.getString("assigned_user"));
+      };
+
+  private final NamedParameterJdbcTemplate jdbcTemplate;
+
+  // -- Common Resource Methods -- //
+
+  @Autowired
+  public ResourceDao(NamedParameterJdbcTemplate jdbcTemplate) {
+    this.jdbcTemplate = jdbcTemplate;
+  }
+
+  public boolean deleteResource(UUID workspaceId, UUID resourceId) {
```

Comment:
Why do we need the workspace ID if the resource ID is the PK? Belt and suspenders approach?